### ISSUE TITLE
va-modal and va-crisis-line-modal Accessibility Bugs (Readers and Focus)

### ISSUE THREAD

**Issue by Andrew565** [department-of-veterans-affairs/vets-design-system-documentation#2255]
## Description
From [this review](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2100).

Must: There were some issues with the crisis line modal, I don't see these issues on the main header in production.
* Crisis line modal is not announced by screen readers.
* You can tab outside of the modal to the VA logo link. This also means that focus is not trapped inside the modal, as it should be (see production's crisis modal).

### Re: CLM not being read out by screen readers.
One of the issues is the `hideOthers` function, which applies the `aria-hidden="true"` prop to all of the DOM elements that are outside of the modal so that they are ignored. This is desired behavior, except that inside of Storybook it also gets applied to the `#docs-root` element, which is a parent of the modal, and effectively makes the modal "invisible" to screen readers. I was able to work around this by adding the following code to the va-modal's `setupModal` function:

```
// "Unhides" the root element for Storybook, needed for screen readers
document.querySelector('#docs-root')?.setAttribute('aria-hidden', 'false');
```

My theory is that because this runs _after_ the hideOthers function the screen reader system is still unable to automatically read out the modal's content because it appears "out of order" with the prescribed events. There is a possibility that adding "aria-live" could help with this, but this is not something I've tried nor am I the most knowledgeable about. Currently, when the modal opens, focus is placed on the "close" button of the modal, and anyone using a screen reader will hear that close button announced, rather than the actual content of the modal. To hear the rest of the modal's content requires using VoiceOver's navigation (VO-Right Arrow) to get to the header and subsequently to the rest of the content.

Another strategy that was tried was adding the `#docs-root` element to the array of `ariaHiddenNodeExceptions`, the first argument of the `hideOthers` function. This was effective at making the screen reader work but caused an error when running tests due to that element not existing in the testing environment.

### Re: Focus trapping.
Keyboard focus is not being properly trapped by the CLM modal when it is being used as part of the Minimal Header. Initial testing suggests that the `focusin` event is not being triggered properly when using the tab key to jump between links. Switching the event to `keyup` allows for the `handleFocus` listener to fire every time the tab key is used, but causes a momentary 'hiccup' where the focus lands outside the modal for a split-second before being redirected back into the modal. Using `keydown` might alleviate this and prevent the hiccup, but I was not able to initially figure out how to determine if the 'next link' that was being jumped to was outside the modal before landing on it, which is why I tried using `keyup` instead, because it could identify the link that was landed on.

## Tasks
- [ ] Fix Crisis Line Modal so that it is properly announced and read out by screen readers
- [ ] Fix va-modal so that it properly 'traps' keyboard focus within the modal

## Acceptance Criteria
- [ ] Crisis Line Modal is properly announced and read out by screen readers
- [ ] va-modal's focus is properly 'trapped' within the modal and no external links get recognized and read out by screen readers

**Comment by caw310** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Platform-Design-System-5f8de67192551b0012ebb802/issues/department-of-veterans-affairs/vets-design-system-documentation/2255?planning-poker) with Zenhub @Andrew565 @ataker @harshil1793 @it-harrison @jamigibbs @micahchiang @nickjg231 @powellkerry @rmessina1010 @rsmithadhoc

**Comment by Andrew565** [comment]
Ryan has noted in #2207 that Firefox doesn't read anything in the modal.

**Comment by Andrew565** [comment]
Focus Trapping was addressed as part of #2207.
